### PR TITLE
[go-gen] Add main.go response struct generation

### PR DESCRIPTION
**Pull request by willjvsmith**
* Adds generation of main.go (non-auth) structs
  * Moves struct gen into own file `GoServiceMainStructGenerator`
* Updates all tests

There's only about 60 lines of actual new code here (mainly`GoServiceMainStructGenerator:69` to EOF), the snippets take up a lot!

### CODE REVIEW

File: src/test/scala/temple/testfiles/simple.temple [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (c-harding) [inline src/test/scala/temple/testfiles/simple.temple:1]: ?

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainStructGenerator.scala [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    )
+
+  private def generateResponseStruct(root: ServiceRoot, operation: CRUD, fields: ListMap[String, String]): String =
+    mkCode.lines(
+      mkCode(
+        s"// ${operation.toString.toLowerCase}${root.name.capitalize}Response contains a",
+        operation match {
+          case Create => s"newly created ${root.name}"
+          case Read   => s"single ${root.name}"
+          case Update => s"newly updated ${root.name}"
```
Comment (jaylees14) [inline src/main/scala/temple/generate/server/go/service/main/GoServiceMainStructGenerator.scala:94]: What happens for a delete? I know we know there's never a request body, but feels like you should still deal with the case

**Review by c-harding — APPROVED** [review]
nice

**Review by jaylees14 — APPROVED** [review]
All looks good to me :)

**Pull request merged**